### PR TITLE
[SC-546] Tests for DeleteEntry function

### PR DESCRIPTION
**Pull request by ngadiyak**

### CODE REVIEW

File: node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+      |func write%f(k: String, v: %t) = {
+      |  [%fEntry(k, v)]
+      |}
+      |""".stripMargin
+
+  val writeEntriesFunc =
+    """
+      |@Callable(i)
+      |func write%sEntries() = {
+      |    FOLD<%s>(a, [], writeEntry)
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala]: you can shorten such functions to one line
`func write%sEntries() = FOLD<%s>(a, [], writeEntry)`

File: node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+import com.wavesplatform.common.utils.{Base58, EitherExt2}
+import com.wavesplatform.it.api.SyncHttpApi._
+import com.wavesplatform.it.sync._
+import com.wavesplatform.it.util._
+import com.wavesplatform.lang.v1.compiler.Terms.{CONST_BOOLEAN, CONST_LONG, CONST_STRING}
+import com.wavesplatform.lang.v1.estimator.v2.ScriptEstimatorV2
+import com.wavesplatform.transaction.smart.SetScriptTransaction
+import com.wavesplatform.transaction.smart.script.ScriptCompiler
+
+class RemoveEntrySuite extends BaseSuite {
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala]: add assertions that you can't read removed entries in the next invocations
`this.getBinaryValue("some-removed-key")`

File: node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+      assertBadRequestAndMessage(
+        miner.waitForTransaction(invokeScript(address, s"delete101Entries", wait = false)),
+        "Error while executing account-script: List size exceed 101"
+      )
+
+      miner.getData(address) should have size 0
+    }
+
+    "Trying of writing key longer than 400 bytes should produce an error" in {
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala]: and removing entry by longer key

File: node-it/src/test/scala/com/wavesplatform/it/account/storage/BaseSuite.scala [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import com.typesafe.config.{Config, ConfigFactory}
+import com.wavesplatform.it.transactions.NodesFromDocker
+import com.wavesplatform.it.{ExternalNode, Node, NodeConfigs}
+import monix.eval.Coeval
+import org.scalatest._
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+class BaseSuite
```
Comment (msmolyakov) [inline node-it/src/test/scala/com/wavesplatform/it/account/storage/BaseSuite.scala]: move to root `it` package

File: node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+import com.wavesplatform.lang.v1.estimator.v2.ScriptEstimatorV2
+import com.wavesplatform.transaction.smart.SetScriptTransaction
+import com.wavesplatform.transaction.smart.script.ScriptCompiler
+
+class RemoveEntrySuite extends BaseSuite {
+
+  val writeEntry =
+    """
+      |@Callable(i)
+      |func write%f(k: String, v: %t) = {
```
[the remark this answers is not in the document]
Comment (ngadiyak) [reply node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala]: ok

File: node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala (hunk 1 above)
Comment (ngadiyak) [reply node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala]: ok

File: node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    val stringTestData  = "String"  -> "someValue"
+    val integerTestData = "Integer" -> 1
+    val booleanTestData = "Boolean" -> true
+
+    for (data <- Seq(stringTestData, integerTestData, booleanTestData)) s"${data._1} entry could be removed from account storage" in {
+      val t = data._1
+      val k = "someKey"
+      val v = data._2
+      val address =
+        createDapp(script + writeEntry.replaceAll("%f", t).replaceAll("%t", if (t.contains("Int")) "Int" else t))
```
[the remark this answers is not in the document]
Comment (ngadiyak) [reply node-it/src/test/scala/com/wavesplatform/it/account/storage/RemoveEntrySuite.scala]: ok

**Pull request merged**